extensions/data-workspace/src/common/workspaceTreeDataProvider.ts

```diff
@@ -24,6 +25,7 @@ export class WorkspaceTreeDataProvider implements vscode.TreeDataProvider<Worksp
 	readonly onDidChangeTreeData?: vscode.Event<void | WorkspaceTreeItem | null | undefined> | undefined = this._onDidChangeTreeData?.event;
 
 	async refresh(): Promise<void> {
+		Logger.log(`Refreshing projects tree`);
```

Comment:
You should probably be logging the end times (with duration) as well. Basically you're just adding a bunch of performance markers (see https://github.com/Microsoft/vscode/wiki/Performance-Issues for examples)

You should also look to see if https://github.com/Microsoft/vscode/wiki/Performance-Issues#profile-the-running-extensions is useful. If they enable profiling and then reload the window does it capture the startup times?

Reply:
Ah yes adding the duration is much easier than doing the math myself from the start and end timestamps :D 

The "Running Extensions" window has all the extension activation times. I tried enabling profiling and restarting the extension host, but it doesn't look like the profiled time includes the time spent in activation. The extension host profile also didn't look that helpful for figuring out where time is getting spent in the data workspace extension.  
![image](https://user-images.githubusercontent.com/31145923/190024824-89087b38-a626-4bfb-974c-6e16c7666f1d.png)